api/index.ts

```diff
@@ -18,8 +21,18 @@ const start = async () => {
   // NOTE: Awaited to ensure `.use` is registered on `fastify`
   await fastify.register(middie);
 
+  // Auth0 plugin
+  void fastify.register(fastifyAuth0, {
+    domain: process.env.AUTH0_DOMAIN,
+    audience: process.env.AUTH0_AUDIENCE
+  });
+  void fastify.register(jwtAuthz);
+
   void fastify.use('/test', testMiddleware);
 
+  // Hooks
+  void fastify.addHook('preValidation', auth0Verify);
+
```

Comment:
I am glad this adds authentication, but I got confused why I could no longer `curl localhost:3000`:

```bash
$ curl localhost:3000
{"statusCode":401,"error":"Unauthorized","message":"Missing Authorization HTTP header."}
````

I realise I can add the header, but I would prefer to not set `auth0Verify` to be blanket on all routes. Soon, we will add `GET /ping`, and that should not be gated.

Reply:
This is expected. We would (like we currently do on the existing API) block all routes by default and then add exceptions to certain routes that do not need auth.

Deny all by default and then make exceptions with middleware.

Reply:
Sounds less fun than my way, but fine.

Reply:
It's easier to audit endpoints that do not need "protection" than those that do. Plus, you are defaulting to zero trust, which is more secure.

Reply:
How do we do that with fastify, though? I haven't found a way to de-register a hook.

Reply:
You can achieve something similar(ish) by having two plugins "protected" and "public" and registering the hook for the "protected" one, but that's not quite the same.